More operator docs (#2107)
kroxylicious/kroxylicious: docs/modules/configuring/con-understanding-virtualkafkacluster-status.adoc

[id='con-understanding-virtualkafkacluster-status-{context}']
= `VirtualKafkaCluster` status

== `ResolvedRefs` conditions

When you create a `VirtualKafkaCluster`, the operator will check whether:

* a `KafkaProxy` corresponding to the `spec.proxyRef` exists,
* a `KafkaProxyIngress` corresponding to each `spec.ingresses` exists, and whether they refer to the same `KafkaProxy` as the virtual cluster,
* any `Secret` referred to in the `tls` property exists.

The operator will update the `status.conditions` with a `ResolvedRefs` condition accordingly.

> .Example `VirtualKafkaCluster` status when all referenced resources exist.

Reviewer: ```suggestion
.Example `VirtualKafkaCluster` status when all referenced resources exist
```

[source,yaml]
----
kind: VirtualKafkaCluster
apiVersion: kroxylicious.io/v1alpha1
metadata:
  # ...
  generation: 12
spec:
  # ...
status:
  observedGeneration: 12 # <1>
  conditions:
    - type: ResolvedRefs # <2>
      status: True # <3>
      observedGeneration: 12
----
<1> Because the status's `observedGeneration` matches the `metadata.generation` we know that this status is up-to-date with respect to the `spec`.

> <2> The `ResolvedRefs` condition type is used for problems to do with the referenced resources.

Reviewer: ```suggestion
<2> The `ResolvedRefs` condition type reports any issues with referenced resources.
```

> <3> In this case the `ResolvedRefs` condition has status `True`, meaning all the referenced resources exist.

Reviewer: ```suggestion
<3> A status value of `True` means that all referenced resources exist. 
```

> A `ResolveRefs` condition with `status: False` means there's a problem with one or more of the referenced resources. In this case the condition will have a `reason` and `message` properties providing more detail about the problem.

Reviewer: ```suggestion
A status value of `False` means that one or more of the referenced resources is missing. In this case, the condition includes `reason` and `message` properties with more details.
```

== `Accepted` conditions

> If a virtual cluster has a valid `spec`, the operator will try to update the proxy to honor it.

Reviewer: ```suggestion
When a `VirtualKafkaCluster` has a valid spec, the operator attempts to configure the proxy instance accordingly.
```

This might not be possible.

> For example, although the `spec` is valid, it might not be compatible with another existing virtual cluster running in the same proxy instance.

Reviewer: ```suggestion
For example, the `spec` may be valid but incompatible with other virtual clusters running in the same proxy instance.
```

> The operator uses `type: Accepted` conditions to signal whether or not a virtual cluster has been successfully configured within a proxy instance.

Reviewer: ```suggestion
The operator sets a condition type of `Accepted` in `status.conditions` to indicate whether or not a virtual cluster has been successfully configured within a proxy instance.
```
